node/src/main/scala/org/bitcoins/node/PeerManager.scala

```diff
@@ -212,11 +220,11 @@ case class PeerManager(
 
     if (finder.hasPeer(peer)) {
       //one of the peers that we tried, failed to init within time, disconnect
-      finder.getData(peer).client.close()
+      finder.getData(peer).client.foreach(_.close())
     } else if (peerData.contains(peer)) {
       //this is one of our persistent peers which must have been initialized earlier, this can happen in case of
       //a reconnection attempt, meaning it got connected but failed to initialize, disconnect
-      peerData(peer).client.close()
+      peerData(peer).client.foreach(_.close())
```

Comment:
This is a `Future` that you are dropping on the floor.

Use `.map`, don't use `foreach` with `Future` as it encourages dropping `Future`'s on the floor.